### DISCUSSION TITLE
MutatorBase<T> not implementing IMutator

### DISCUSSION THREAD

**Discussion by michalusio** [stryker-mutator/stryker-net#2582] [General]
Hello!

I have searched for that question being asked here previously, but found nothing.

I am implementing a new mutator and I see that the abstract class MutatorBase<T> does not implement IMutator - you have to implement both in the mutator.

Using MutatorBase<T> without implementing IMutator doesn't really make sense to me, was it a design decision to separate those two?

If discussions are not a place for this question, please delete it 😃

**Comment by dupdob** [comment]
I can not really justify this, I have no clear idea. This may be related to support custom mutators as dynamic add-in/plug-in in the feature: the interface can be used as a marker do detect mutator classes at run time.
